### PR TITLE
feat(rxTags): Add rxTags component.

### LINKED ISSUE #918: Create 'tags' component
Here's the look/feel for tags:

tags

Interaction design for this is based on https://material.angularjs.org/#/demo/material.components.chips, which should be used as a base for building a tags component, since it provides good flexibility with read-only functionality, accommodates hierarchy through use of categories, etc. 

The main thing not provided by Material is a type-ahead drop down, but after consulting with @TheNando it sounds like that won't be too difficult to add and customize. See https://github.com/rackerlabs/encore-ui/issues/909, which should be used by default for this component.

### PR DESCRIPTION
**Pull request by megalord**
Fixes #918 

- [x] Dev LGTM
- [ ] Dev LGTM
- [x] Design LGTM

screen shot 2015-08-14 at 1 56 35 am

@glynnis This does not include the feature to show the list on click.  I started down that path, and it's a bit complex.  I am also going to be out next week, so someone else can have a go or I will look at it again down the road.

### CODE REVIEW

File: src/rxTags/rxTags.exercise.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    return function () {
+        var component, tag, numTags;
+
+        before(function () {
+            if (options.instance !== undefined) {
+                component = options.instance;
+            } else {
+                component = rxTags.main;
+            }
+            component.count.then(function (num) {
```
Comment (Droogans) [inline src/rxTags/rxTags.exercise.js]: I know `count` is a property, but it should mirror protractor's `count()` function.

Every other component does this. `:/`

Comment (megalord) [inline src/rxTags/rxTags.exercise.js]: oops my bad

File: src/rxTags/templates/rxTags.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div class="rx-tags" ng-click="focusInput($event)">
+    <div class="tag"
+        ng-repeat="tag in tags track by tag.text"
+        ng-keydown="removeIfBackspace($event, tag)"
+        tabindex="0">
+        <i class="fa fa-tag"></i>
+        <span class="text">{{tag.text}}</span>
+        <span class="category">({{tag.category}})</span>
+        <i class="fa fa-times" ng-click="remove(tag)"></i>
+    </div>
```
Comment (CITguy) [inline src/rxTags/templates/rxTags.html]: Can we make this div an `<rx-tag>` element directive?  I can see developers wanting to display these tags outside of a form.

Comment (megalord) [inline src/rxTags/templates/rxTags.html]: How would that differ from using the `disabled` attribute?  Even then, I suppose it's nice not to have to create an `ng-model`.

Comment (CITguy) [inline src/rxTags/templates/rxTags.html]: nevermind, I think I misinterpreted what the directive does

Comment (CITguy) [inline src/rxTags/templates/rxTags.html]: However, it probably wouldn't be a bad idea to add a disabled state example to the demo.

**Comment by megalord** [comment]
The only test failure is that brittle test for `rxNotify` where a stale element error is thrown.

**Comment by Droogans** [comment]
@CITguy rebuilding the Travis job to see what happens when the visual regression tests see the exact same SHA and the same branch with an existing PR open against it.

My guess is that it'll fail to push it up? I'm not sure what'll happen.

**Comment by Droogans** [comment]
@megalord can you please push up a new commit?

All you need to do is

```bash
git commit --amend
Waiting for ${EDITOR}...
# save, and close, no changes
git push -f [origin|upstream] rx-tags
```

**Comment by CITguy** [comment]
LGTM

**Comment by glynnis** [comment]
Design LGTM. @CITguy would you mind creating a story for the remaining work in a way that will make sense to devs? Here's what's left:

> We also need a version of rxTags that reveals the list on click (like the second example on 
> http://rackerlabs.github.io/encore-ui/#/components/typeahead). Requiring the user to type should be 
> the default, but it would be nice to allow consumers of the framework an option to use the second 
> example.

**Comment by CITguy** [comment]
@glynnis, https://jira.rax.io/browse/FRMW-207

@megalord, please squash

**Pull request merged**